src/main/java/com/klodnicki/Bike/v2/service/RentBikeService.java

```diff
@@ -0,0 +1,187 @@
+package com.klodnicki.Bike.v2.service;
+
+
+import com.klodnicki.Bike.v2.DTO.bike.BikeForNormalUserResponseDTO;
+import com.klodnicki.Bike.v2.DTO.rent.RentRequestDTO;
+import com.klodnicki.Bike.v2.DTO.rent.RentResponseDTO;
+import com.klodnicki.Bike.v2.DTO.user.UserForNormalUserResponseDTO;
+import com.klodnicki.Bike.v2.model.RentRequest;
+import com.klodnicki.Bike.v2.model.entity.Bike;
+import com.klodnicki.Bike.v2.model.entity.ChargingStation;
+import com.klodnicki.Bike.v2.model.entity.Rent;
+import com.klodnicki.Bike.v2.model.entity.User;
+import com.klodnicki.Bike.v2.repository.BikeRepository;
+import com.klodnicki.Bike.v2.repository.ChargingStationRepository;
+import com.klodnicki.Bike.v2.repository.RentRepository;
+import com.klodnicki.Bike.v2.repository.UserRepository;
+import jakarta.transaction.Transactional;
+import org.modelmapper.ModelMapper;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class RentBikeService implements RentBikeGenericService {
+
+    private final BikeRepository bikeRepository;
+    private final RentRepository rentRepository;
+    private final UserRepository userRepository;
+    private final ChargingStationRepository chargingStationRepository;
```

Comment:
why shall RentRepository be deleted from here? In fact it is a RentBikeService class -> why it cannot have access straight to RentRepository please tell me?

Reply:
RentRepository is fine now, as you don't have separate service which access it. I mean you could have for that exact reason, but it's fine for now to leave it that way. Regarding other repositories it is due to the exact same reason -> because you have service which is accessing this repo.